ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -668,4 +694,64 @@ public JoinKey rewriteWith(
       return this;
     }
   }
+
+  private static final class ForeignJoinKey implements JoinKey {
+    private final ColumnName keyColumn;
```

Comment:
What is the purpose of this key column? It looks like the only place it's being used is in `resolveKeyName()`. This method should never be called for foreign key joins, since we don't need to resolve a key name -- the key column names are exactly the names of the key columns for the left source.